backend/src/main/kotlin/fr/gouv/cnsp/monitorfish/domain/use_cases/prior_notification/GetPriorNotifications.kt

```diff
@@ -147,23 +152,44 @@ class GetPriorNotifications(
         return enrichedPaginatedList
     }
 
+    private fun PaginatedList<PriorNotification, PriorNotificationStats>.getReportings(): List<Reporting> {
+        val internalReferenceNumbers = data.mapNotNull { it.vessel?.internalReferenceNumber }
+        val reportings =
+            reportingRepository.findAll(
+                ReportingFilter(
+                    vesselInternalReferenceNumbers = internalReferenceNumbers,
+                    isArchived = false,
+                    isDeleted = false,
+                    types = listOf(ReportingType.INFRACTION_SUSPICION),
+                ),
+            )
+
+        return reportings
+    }
+
     private fun enrichPriorNotificationsWithVessel(
         priorNotifications: List<PriorNotification>,
     ): List<PriorNotification> {
         val vesselsViaVesselsIds =
             priorNotifications
+                .asSequence()
                 .filter { it.isManuallyCreated }
                 .mapNotNull { it.logbookMessageAndValue.logbookMessage.vesselId }
+                .distinct()
```

Comment:
On re met pas dans la clause `IN` les doublons, ça permet d'avoir moins de données à transiter en IN/OUT